modules/tinymce/src/themes/silver/main/ts/modes/Inline.ts

```diff
@@ -124,6 +137,18 @@ const render = (editor: Editor, uiComponents: RenderUiComponents, rawUiConfig: R
   };
 };
 
+const getApi = (editor): InlineApi => ({
+  ui: {
+    show: () => {
+      editor.fire('show');
```

Comment:
I don't think we can do this, as `show` and `hide` have different meanings/are used already to indicate the entire editor is shown/hidden. That is likely to break certain integrations, since they'd be expecting the editor to no longer be interact-able.

As such, we likely need to consider how these APIs are going to work in more detail, or come up with new event names for this. I'm leaning towards the former, as this really needs access to the items within `render` and that's likely to become a higher priority if we keep adding new APIs.

Reply:
One thought I had just because of the comment in `Theme.ts`, was if this was moved to the `editor.ui` namespace, then get could have the `renderUI` function return an API to bind to that. That would however mean we'd either end up with an inconsistent API for inline vs iframe, or we'd end up with the API being `editor.ui.inline.show()` which we sort of wanted to avoid. In saying that though, either API, including what we'd originally planned here will always have some form of inconsistency as `editor.theme.inline` won't do anything for iframe mode. Anyways, needless to say we probably didn't think this through properly and need to revaluate a bit sorry Simon!

I wonder if we need to try and come up with a common API that applies to both. That would then allow something like `editor.ui.show()` and it could just be a noop/console log for iframe maybe? (Note: my notes here are just arbitrary thoughts to try and help get a discussion going, so take them as you will)

Reply:
PRs are for wider discussions as well:)
The problem of hooking into the items within `render` was something I encountered, and like you say, we're probably going need to do the same with more things in the future. So if we were to go ahead with your last idea, would perhaps `editor.ui.show()`/`editor.ui.hide()`/etc. be a part of the Editor (noop by default) and then a theme would optionally choose to override these? Maybe just for inline mode in silver to start with.

I've added a commit that does this. Let me know if it looks reasonable or if you think we should try another approach.

Reply:
I'm okay with this approach; can't think of anything better atm. We should definitely take a bit more time to think about/discuss it though. Do we want to organise a call for some time next week?

Reply:
If we still feel a bit uncertain we could do that, especially since this isn't just about the specific APIs added in this PR but also about the new namespace which we would create(or not create)